test/Middleware/AuthHttpBasicTest.php

```diff
@@ -27,19 +27,71 @@ protected function getMiddleware()
             $this->registry
         );
     }
+   
 
-    public function testNotAuthenticated()  //test ob user nicht authentifiziert
+    public function testNotAuthenticatedWithoutHeader()  //test ob user nicht authentifiziert
     {
         $username = 'testUser01';
+        
+        $this->authDriver->method('authenticate')->willReturn(false); //not authenticated
+        $this->registry->method('getAuth')->willReturn($username);
+        $request = $this->requestFactory->createServerRequest('GET', '/test');
+        $middleware = $this->getMiddleware();
+        $response = $middleware->process($request, $this->handler);
+        
+        $noAuthHeader = $this->recentlyHandledRequest->getAttribute('NO_AUTH_HEADER'); //ohne authorization header
+     
+        $notAuth=$this->recentlyHandledRequest->getAttribute('HORDE_AUTHENTICATED_USER'); //nicht authorisiert->willreturn(false)
+        
+        $this->assertNull($notAuth); //not authenticated
+        $this->assertEquals($username, $noAuthHeader); 
+       
+        
+      
+    }
+    
+    public function testNotAuthenticatedWithHeader()  //test ob user nicht authentifiziert
+    {
+        $username = 'testUser01';
+        
         $this->authDriver->method('authenticate')->willReturn(false);
         $this->registry->method('getAuth')->willReturn($username);
         $request = $this->requestFactory->createServerRequest('GET', '/test');
         $middleware = $this->getMiddleware();
         $response = $middleware->process($request, $this->handler);
+        
+        $AuthHeader = !$this->recentlyHandledRequest->getAttribute('NO_AUTH_HEADER'); //mit authorization header
```

Comment:
This does not make much sense if you assertFalse for this later. If you want to make sure it was not set, you just get it here and `assertNull` later.

Reply:
Also first letter of variable names should be lowercase